### ISSUE TITLE
Exporting timestamped log files

### ISSUE THREAD

**Issue by snnd** [risu-io/neo4j-luxury-docker#2]
For regular automated backups it would be nice to have a script, that creates a timestamped snaptshot of the graphdb from inside the data container.

**Comment by hrcl** [comment]
does the containers log to stdout? (docker-compose logs, docker logs $container_id)
I would probably use fluentd (http://www.fluentd.org/guides/recipes/docker-logging) with automated conf file building with docker-gen (http://jasonwilder.com/blog/2014/03/17/docker-log-management-using-fluentd/)

**Comment by snnd** [comment]
Hello @hrcl,

thank you for participating! :+1: 

the neo4j container produces some stdout logging but also creates a log file of it's own in /data/neo4j/data/log. there are some options i could imagine to be interesting:

1. just extract the the log file from the data container. i am not sure if the std out logging contains all vital log options yet, or if it is even capable of logging more vital options. this is to be proven.

2. keeping a distributed approach in mind, it could be a great idea to collect the logs, neo4j currently puts into the logfile, and pipe it to some other service like logstash or similiar.

3. some combination of the above. perhabs configure neo4j to log into stdout and then collect the logs somewhere else in a logging service. 

is tthe 3rd option what you would in mind with fluentd?

Beste regards,
Dennis

**Comment by hrcl** [comment]
Although not being a 12factor extremist, [those folks are right when it comes to logging](http://12factor.net/logs).
Having said that i just meta-cowboy coded (*) [an approach where you create those logfiles as a named pipe](https://github.com/hrcl/stdoutpipe) to let them log to stdout by another container ;)

Having all container output on stdout allows to collect those logs either on the host or by passing it to a syslog service (docker run --log-driver=syslog), which was introduced with version 1.6), although syslog alone gives no further insight. 

**tl;dr**: all stuff should go to stdout. Then choose logstash, fluentd, syslog, or something else for post-processing.

(\*) *meta* as in: having not (yet) tested if it runs at all
